tests/core/snapd-refresh-vs-services/task.yaml

```diff
@@ -0,0 +1,212 @@
+      http --quiet --download --output snapd_2.49.2.snap GET "$SNAPD_2_49_2_ARMHF"
+    fi
+
+    echo "Refreshing snapd to 2.49.2"
+    snap install --dangerous snapd_2.49.2.snap
+    snap version | MATCH 2.49.2
+
+  elif [ "${SNAPD_VERSION_UNDER_TEST}" = "pr" ]; then
+      # refresh back to the version we originally had from before the test 
+      # started
+      echo "Refreshing snapd to version from the pr"
+      snap install --dangerous snapd-pr.snap
+  fi
+
+  # now install another service that will either get Requires= or Wants= for
+  # usr-lib-snapd.mount, depending on the variant of the test
+  "$TESTSTOOLS"/snaps-state install-local-as test-snapd-simple-service test-snapd-simple-service_alt
+
+  # check that it is still initially active
+  snap services|MATCH ".*test-snapd-simple-service_alt.test-snapd-simple-service\s*enabled\s*active.*"
+
+  # if we are not running with 2.49.2, then the second service should have been
+  # generated with Wants=, but if we are doing 2.49.2 then the second service 
+  # should have Requires= in it
+  if [ "${SNAPD_VERSION_UNDER_TEST}" = "2.49.2" ];then 
+    MATCH Requires=usr-lib-snapd.mount < /etc/systemd/system/snap.test-snapd-simple-service_alt.test-snapd-simple-service.service
+  else
+    MATCH Wants=usr-lib-snapd.mount < /etc/systemd/system/snap.test-snapd-simple-service_alt.test-snapd-simple-service.service
+  fi
+
+  # now repack current snapd to refresh to it again and observe the results
+  unsquashfs -d ./snapd-unpacked snapd-pr.snap
+  snap pack --filename=snapd_repacked.snap snapd-unpacked
+
+  # save the PID's of the services
+  systemctl show -p MainPID snap.test-snapd-simple-service.test-snapd-simple-service > old-main.pid
+  systemctl show -p MainPID snap.test-snapd-simple-service_alt.test-snapd-simple-service > old-main_alt.pid
+
+  echo "Refresh snapd"
+  snap install --dangerous snapd_repacked.snap
+
+  # for all variants of the test, the main.pid should be the same since it did
+  # not ever contain the Requires=
+  systemctl show -p MainPID snap.test-snapd-simple-service.test-snapd-simple-service > new-main.pid
+
+  if [ "$(cat new-main.pid)" != "$(cat old-main.pid)" ]; then
+    echo "The service without Requires= was restarted, test is broken"
+    exit 1
+  fi
+
+  # for the variants of the test that have the fix in them, that is stable and
+  # and pr, then we also should not have had the alt service pid change due to a
+  # restart
+  # for the other variant (2.49.2 only), we unfortunately ended up needing to
+  # restart the service so it should have a different PID, but it should be 
+  # running again
+
+  systemctl show -p MainPID snap.test-snapd-simple-service_alt.test-snapd-simple-service > new-main_alt.pid
+  if [ "${SNAPD_VERSION_UNDER_TEST}" = "2.49.2" ]; then
+    if [ "$(cat new-main_alt.pid)" = "$(cat old-main_alt.pid)" ]; then
+      echo "Somehow the service was not killed as expected ... test is probably broken"
+      exit 1
+    fi
+  else
```

Comment:
elif could be used ads well

Reply:
personally I think this is more readable